{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6e7cc487-ef39-4ec0-810e-701a49ecbbb1",
   "metadata": {},
   "source": [
    "# Inquiries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d7b0191c-2312-412f-b8f0-5d37f37e140b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make some helpful functions \n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import glob\n",
    "from functools import partial \n",
    "from os import chdir\n",
    "import io\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "\n",
    "def submitTerminal(command, printing=False, message=''):\n",
    "    # quick command to submit jobs to terminal\n",
    "    start = time.time()\n",
    "    res=subprocess.run(command.split(' '), stdout=subprocess.PIPE)\n",
    "    end = time.time()\n",
    "    sys.stdout.write('EXEC_TIME in sec: '+ str(round(end - start, 3)) + ' : ')\n",
    "    if printing:\n",
    "        print(res.stdout.decode('utf-8'))\n",
    "    if message=='':\n",
    "        return(res.stdout.decode('utf-8'))\n",
    "    else:\n",
    "        print(message, '\\n')\n",
    "        \n",
    "def plink2_cmd(plink_file_path, output_path, options):\n",
    "    cmd = f'/data/CARD/PD/AMP-PD/Plink/plink2_dev_June8/plink2 --pfile {plink_file_path} {options} --out {output_path} --threads 2'\n",
    "    return cmd\n",
    "        \n",
    "def plink_cmd(plink_file_path, output_path, options):\n",
    "    cmd = f'/data/CARD/PD/AMP-PD/Plink/plink2_dev_June8/plink2 --bfile {plink_file_path} {options} --out {output_path} --threads 2'\n",
    "    return cmd\n",
    "\n",
    "def plink1_cmd(plink_file_path, output_path, options):\n",
    "    cmd = f'plink --bfile {plink_file_path} {options} --out {output_path} --threads 2'\n",
    "    return cmd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d3386db-2580-47cf-949e-671099ee9f42",
   "metadata": {},
   "source": [
    "## LRRK2 R1441G/C\n",
    "From Andy 11/29/2021\n",
    "\n",
    "\"do you happen to know if there are any R1441C mutation carriers in PPMI – I’m pretty sure there are R1441G, but don’t know about the C allele\"\n",
    "* R1441G: chr12:40310434:C:G\n",
    "* R1441C: chr12:40310434:C:T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "ea73bd1c-86d5-44c2-95a2-7d87d4d04bd7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EXEC_TIME in sec: 10.199 : extract snps of interest from the PLINK2 AMP v2.5 files \n",
      "\n"
     ]
    }
   ],
   "source": [
    "plink_file_path = '/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/formatted_split_normalized_pfile/reformatted_split_normalized_chr12'\n",
    "\n",
    "t = submitTerminal(plink2_cmd(f'{plink_file_path}', 'R1441', f'--snps chr12:40310434:C:T,chr12:40310434:C:G --export A'), message='extract snps of interest from the PLINK2 AMP v2.5 files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "685c1ecd-88b9-487f-94cc-5212017cec3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>chr12:40310434:C:G_C</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:40310434:C:T_C</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>91.0</td>\n",
       "      <td>10316.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "chr12:40310434:C:G_C     1        2\n",
       "chr12:40310434:C:T_C               \n",
       "1                      NaN     11.0\n",
       "2                     91.0  10316.0"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = pd.read_csv('R1441.raw', sep='\\t')\n",
    "d.pivot_table(index='chr12:40310434:C:T_C', columns=['chr12:40310434:C:G_C'], values='IID', aggfunc='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "7181c3e4-c08e-4cbe-acde-dc22575a8c61",
   "metadata": {},
   "outputs": [],
   "source": [
    "d['R1441G'] = 2-d['chr12:40310434:C:G_C']\n",
    "d['R1441C'] = 2-d['chr12:40310434:C:T_C']\n",
    "d['DATASET'] = [x[:2] for x in d.IID]\n",
    "d = d.drop(columns=['PAT', 'MAT', 'SEX', 'PHENOTYPE', 'chr12:40310434:C:G_C', 'chr12:40310434:C:T_C'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "66047126-6ce1-4ed1-b25e-a217c7763aa2",
   "metadata": {},
   "outputs": [],
   "source": [
    "dcov = pd.read_csv('/data/CARD/PD/AMP-PD/release2.5_COVFILE/AMPPD_releasev2.5_covariates_JUNE_2021.csv')\n",
    "df = pd.merge(d, dcov, left_on='IID', right_on='ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "41f52375-fc97-4128-a51b-7ccfb556e3aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>R1441G</th>\n",
       "      <th colspan=\"2\" halign=\"left\">0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1441C</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DATASET</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BF</th>\n",
       "      <td>171</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HB</th>\n",
       "      <td>1173</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LB</th>\n",
       "      <td>4579</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LC</th>\n",
       "      <td>542</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PD</th>\n",
       "      <td>1495</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP</th>\n",
       "      <td>1769</td>\n",
       "      <td>1</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SU</th>\n",
       "      <td>259</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SY</th>\n",
       "      <td>328</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "R1441G      0      1\n",
       "R1441C      0  1   0\n",
       "DATASET             \n",
       "BF        171  1   0\n",
       "HB       1173  0   0\n",
       "LB       4579  0   0\n",
       "LC        542  4  53\n",
       "PD       1495  4   1\n",
       "PP       1769  1  37\n",
       "SU        259  0   0\n",
       "SY        328  1   0"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d.pivot_table(index='DATASET', columns=['R1441G', 'R1441C'], values='IID', aggfunc='count', fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "803b3ff7-dd4c-4ef2-aa26-3c378a25e9f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['dataset'] = [x if pd.isna(a) else x if 'Genetic' not in a else f'PP-{a}' for x, a in zip(df.DATASET, df.ENROLL_STUDY_ARM)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "1e70584b-8736-47b7-9227-8b2d3fa23233",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>R1441G</th>\n",
       "      <th colspan=\"2\" halign=\"left\">0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1441C</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BF</th>\n",
       "      <td>171</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HB</th>\n",
       "      <td>1173</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LB</th>\n",
       "      <td>4579</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LC</th>\n",
       "      <td>542</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PD</th>\n",
       "      <td>1495</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP</th>\n",
       "      <td>811</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP-Genetic Cohort PD</th>\n",
       "      <td>245</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP-Genetic Cohort Unaffected</th>\n",
       "      <td>378</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP-Genetic Registry PD</th>\n",
       "      <td>142</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PP-Genetic Registry Unaffected</th>\n",
       "      <td>193</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SU</th>\n",
       "      <td>259</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SY</th>\n",
       "      <td>328</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "R1441G                             0      1\n",
       "R1441C                             0  1   0\n",
       "dataset                                    \n",
       "BF                               171  1   0\n",
       "HB                              1173  0   0\n",
       "LB                              4579  0   0\n",
       "LC                               542  4  53\n",
       "PD                              1495  4   1\n",
       "PP                               811  0   0\n",
       "PP-Genetic Cohort PD             245  1  16\n",
       "PP-Genetic Cohort Unaffected     378  0  16\n",
       "PP-Genetic Registry PD           142  0   2\n",
       "PP-Genetic Registry Unaffected   193  0   3\n",
       "SU                               259  0   0\n",
       "SY                               328  1   0"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.pivot_table(index='dataset', columns=['R1441G', 'R1441C'], values='IID', aggfunc='count', fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "5d83d80b-fa7a-4330-9fa3-03ba2cbc62f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.IID[(df.DATASET=='PP')&(df.R1441C==1)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d8f75e4-0579-47a6-9fa1-b74db8bcbdc0",
   "metadata": {},
   "source": [
    "## Create Polygenic Risk Scores\n",
    "\n",
    "From MichaelB and Kalpana\n",
    "12/12/2021\n",
    "\n",
    "\"Could you send separate scores excluding only the G2019S (I didn’t see 1441C/G in your PRS list) or only the GBA N370S variant? In other words, do not exclude both simultaneously.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "4bfc378c-f5d7-4211-b28e-6047888c74b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# AMP-PD v2.5 plink file reduced to 90 risk associated SNPs\n",
    "## Please refer to the folloing path for more detail\n",
    "## '/data//CARD/PD/AMP-PD/Plink/2021_v2_5release/prs_AMPv2.5/PRS_calc_distribution_forAMPv2.5.html'\n",
    "pfile_path='/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/prs_AMPv2.5/meta5-v3'\n",
    "\n",
    "# Score file (defining the score per allele)\n",
    "score_path = '/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/prs_AMPv2.5/META5_GRS_rsID-replaced2SNPs_wSNPsinLD.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5ae6caae-5a3a-4dcd-8762-d9c9118cb8f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "sc = pd.read_csv(score_path, sep='\\t', header=None)\n",
    "sc.head()\n",
    "sc_excl_g2019s=sc[sc[0]!='rs34637584'].copy()\n",
    "sc_excl_n370s=sc[sc[0]!='rs76763715'].copy()\n",
    "sc.to_csv('score.txt', header=False, index=False, sep='\\t')\n",
    "sc_excl_g2019s.to_csv('score_noG2019S.txt', header=False, index=False, sep='\\t')\n",
    "sc_excl_n370s.to_csv('score_noN370S.txt', header=False, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "18aa1b87-e921-42a9-b4f6-154dc913cac7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EXEC_TIME in sec: 0.088 : extract all META5 variants from the PLINK2 AMP v2.5 files \n",
      "\n",
      "EXEC_TIME in sec: 0.031 : extract all META5 variants from the PLINK2 AMP v2.5 files \n",
      "\n",
      "EXEC_TIME in sec: 0.033 : extract all META5 variants from the PLINK2 AMP v2.5 files \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# create prs\n",
    "plink_file_path = '/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/pfile/amp_v2.5_allchr'\n",
    "for score_path in ['score.txt', 'score_noG2019S.txt', 'score_noN370S.txt']:\n",
    "    t = submitTerminal(plink2_cmd(f'{pfile_path}', f'{score_path.replace(\".txt\",\"\")}',\n",
    "                                  f'--score {score_path}'), \n",
    "                       message='extract all META5 variants from the PLINK2 AMP v2.5 files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "6242bb69-36a6-44e8-943b-9118fc2eb858",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1807, 16)"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Join with Global PCs\n",
    "d1 = pd.read_csv('score.sscore', index_col = 'IID', sep='\\t')\n",
    "d2 = pd.read_csv('/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/euro_king_pca_v2.5_July2021/genetic_ancestry_all_pca.csv', index_col='IID')\n",
    "d = pd.concat([d1,d2],axis=1)\n",
    "\n",
    "# They only need PPMI samples\n",
    "d = d[d.index.str.contains('PP-')].copy()\n",
    "d.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "6987faa6-1f00-4421-a8a9-68a8d56154b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "d.to_csv('score_pcs.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "7fcce03b-226a-443f-a4f3-1eadf7e08e82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"5\" halign=\"left\">dummy</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>InfPop</th>\n",
       "      <th>0</th>\n",
       "      <th>ADMIX</th>\n",
       "      <th>AFRICA</th>\n",
       "      <th>ASIA</th>\n",
       "      <th>EUROPE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APPRDX</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>37.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>168.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       dummy                         \n",
       "InfPop     0 ADMIX AFRICA ASIA EUROPE\n",
       "APPRDX                               \n",
       "1        7.0   NaN    1.0  1.0    6.0\n",
       "2        2.0   NaN    NaN  NaN    1.0\n",
       "3        1.0   NaN    NaN  NaN    1.0\n",
       "4        1.0   NaN    NaN  NaN    NaN\n",
       "5       21.0   NaN    NaN  NaN   76.0\n",
       "6       37.0   1.0    NaN  NaN  168.0"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# chck the \"missing\" PRS\n",
    "mis = pd.read_csv('PATNOs Without PRS90 03JAN22.csv')\n",
    "mis['IID'] = 'PP-' + mis.PATNO.astype('str')\n",
    "mis = mis.set_index('IID').copy()\n",
    "mis['dummy']=1\n",
    "mis.join(d, how='left').fillna(0).pivot_table(index='APPRDX', columns=['InfPop'], values=['dummy'], aggfunc='count')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "439cfefd-4e0f-46a4-a85a-2add13c87b36",
   "metadata": {
    "tags": []
   },
   "source": [
    "## GBA,LRRK2 status and UPSIT\n",
    "From Ken 1/3/2022\n",
    "\n",
    "\"I need a listing by subject including Age, Sex, Raw UPSIT score (number correct), Baseline total UPDRS, Genetic variant (or sporadic PD)\"\n",
    "\n",
    "        'rs2236288',  # GBA E326K\n",
    "        'rs76763715',  # GBA N370N\n",
    "        'rs34637584',  # LRRK2 G2019S\n",
    "        'rs104893877',   # SNCA A53T\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "610f7cab-9e9a-4330-b890-ddcd580f0594",
   "metadata": {},
   "outputs": [],
   "source": [
    "# UPSIT (in LC, sometimes measured multiple times.)\n",
    "d1 =  pd.read_csv(\"/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/UPSIT.csv\").drop_duplicates()\n",
    "\n",
    "# Demographics, enrollment category case-control status\n",
    "d2 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/Demographics.csv').drop_duplicates()\n",
    "d3 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/Enrollment.csv').drop_duplicates()\n",
    "d4 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/amp_pd_case_control.csv').drop_duplicates()\n",
    "t = pd.merge(d1[['participant_id','visit_name', 'visit_month', 'upsit_total_score']], d2[['participant_id', 'age_at_baseline', 'sex']], on='participant_id', how='left')\n",
    "t2 = pd.merge(t, d3[['participant_id', 'study_arm', 'prodromal_category']], on='participant_id',  how='left')\n",
    "t3 = pd.merge(t2, d4, on='participant_id',  how='left').drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "20aa1f83-ebba-4768-9cd9-51060d589c4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# age at diagnosis (at visit_month==0)\n",
    "d5 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/PD_Medical_History.csv').drop_duplicates()\n",
    "d5 = d5[(pd.notna(d5.age_at_diagnosis))&(d5.visit_month==0)].copy()\n",
    "t4 = pd.merge(t3, d5[['participant_id', 'age_at_diagnosis']], on='participant_id',  how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "af9d9225-0a06-4fa1-bfae-414e64c2826a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/Anaconda/envs/py3.8/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (1) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "/usr/local/Anaconda/envs/py3.8/lib/python3.8/site-packages/pandas/core/reshape/merge.py:1214: UserWarning: You are merging on int and float columns where the float values are not equal to their int representation\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# MDS-UPDRS or originalUPDRS\n",
    "## Most of LC and some PD recorded oridinal UPDRS rather than MDS\n",
    "\n",
    "# MDS-UPDRS I\n",
    "d6 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/MDS_UPDRS_Part_I.csv').drop_duplicates()\n",
    "t5 = pd.merge(t4, d6[['participant_id',  'visit_name', 'visit_month', 'mds_updrs_part_i_summary_score']], \n",
    "              on=['participant_id', 'visit_name',  'visit_month'], how='left')\n",
    "\n",
    "# MDS-UPDRS II\n",
    "d7 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/MDS_UPDRS_Part_II.csv').drop_duplicates()\n",
    "t6 = pd.merge(t5, d7[['participant_id',  'visit_name', 'visit_month', 'mds_updrs_part_ii_summary_score']], \n",
    "              on=['participant_id', 'visit_name',  'visit_month'], how='left')\n",
    "\n",
    "# MDS-UPDRS III\n",
    "d8 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/MDS_UPDRS_Part_III.csv').drop_duplicates()\n",
    "t7 = pd.merge(t6, d8[['participant_id',  'visit_name', 'visit_month', 'mds_updrs_part_iii_summary_score', 'upd23b_clinical_state_on_medication']],  \n",
    "              on=['participant_id', 'visit_name',  'visit_month'], how='left')\n",
    "\n",
    "# MDS-UPDRS IV\n",
    "d9 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/MDS_UPDRS_Part_IV.csv').drop_duplicates()\n",
    "t8 = pd.merge(t7, d9[['participant_id', 'visit_name', 'visit_month', 'mds_updrs_part_iv_summary_score']],\n",
    "              on=['participant_id', 'visit_name',  'visit_month'], how='left')\n",
    "\n",
    "# original UPDRS part I/II/III/IV\n",
    "d10 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/UPDRS.csv').drop_duplicates()\n",
    "t9 = pd.merge(t8, d10[['participant_id', 'visit_name', 'visit_month', 'updrs1_ment_behav_mood_score','updrs2_adl_score',\n",
    "                       'updrs3_motor_examination_score','updrs4_therapy_complications_score',\n",
    "#                        'updrs1_on_medication', 'updrs2_on_medication'\n",
    "                      ]],\n",
    "              on=['participant_id', 'visit_name',  'visit_month'], how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "94d6be5f-8433-43f2-8486-d74161a90e8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATASET</th>\n",
       "      <th>LC</th>\n",
       "      <th>PD</th>\n",
       "      <th>PP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>visit_month</th>\n",
       "      <th>visit_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>-1</th>\n",
       "      <th>SC</th>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">0</th>\n",
       "      <th>M0</th>\n",
       "      <td>533.0</td>\n",
       "      <td>1349.0</td>\n",
       "      <td>1827.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M0#2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <th>M6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <th>M12</th>\n",
       "      <td>80.0</td>\n",
       "      <td>556.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">24</th>\n",
       "      <th>M24</th>\n",
       "      <td>96.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M24#2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M24#3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">36</th>\n",
       "      <th>M36</th>\n",
       "      <td>95.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>118.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M36#2</th>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <th>M42</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <th>M48</th>\n",
       "      <td>90.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <th>M54</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">60</th>\n",
       "      <th>M60</th>\n",
       "      <td>37.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M60#2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <th>M72</th>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <th>M84</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>151.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <th>M96</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>159.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <th>M108</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "DATASET                    LC      PD      PP\n",
       "visit_month visit_name                       \n",
       "-1          SC            8.0     NaN     1.0\n",
       " 0          M0          533.0  1349.0  1827.0\n",
       "            M0#2          1.0     NaN     NaN\n",
       " 6          M6            NaN     NaN     3.0\n",
       " 12         M12          80.0   556.0     5.0\n",
       " 24         M24          96.0   489.0   129.0\n",
       "            M24#2         2.0     NaN     NaN\n",
       "            M24#3         1.0     NaN     NaN\n",
       " 36         M36          95.0   391.0   118.0\n",
       "            M36#2         6.0     NaN     NaN\n",
       " 42         M42           NaN     NaN    17.0\n",
       " 48         M48          90.0   121.0    44.0\n",
       " 54         M54           NaN     NaN     4.0\n",
       " 60         M60          37.0    38.0    12.0\n",
       "            M60#2         1.0     NaN     NaN\n",
       " 72         M72           8.0     NaN    61.0\n",
       " 84         M84           NaN     NaN   151.0\n",
       " 96         M96           NaN     NaN   159.0\n",
       " 108        M108          NaN     NaN    29.0"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = t9.copy()\n",
    "d['DATASET'] = d.participant_id.str.slice(0,2) # LD, PP, PD\n",
    "d.pivot_table(index=['visit_month','visit_name'], columns='DATASET', values='participant_id', aggfunc='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "0dbdddc9-a719-4f5d-8683-dc4a3461569c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# reduce to visit_name==0\n",
    "d = d[d.visit_name.isin(['M0', 'M12','M24','M36', 'M48'])].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "70d25a25-f7d6-4992-831f-e06a93bf432e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5923, 23)"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "bbaf9e92-1d6d-4e4b-8640-e0a35adb32e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5923, 23)"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d.drop_duplicates(subset=['participant_id', 'visit_name']).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "a46c504c-0caa-4949-b2cc-6cada6ebd549",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract the META5 variants and make some tiny PLINK binaries \n",
    "snpmaps = {0:['rs2230288','T'],  # GBA E326K\n",
    "        1:['rs76763715','C'],  # GBA N370N\n",
    "        2:['rs34637584','A'],  # LRRK2 G2019S\n",
    "        3:['rs104893877','T'],   # SNCA A53T\n",
    "          }\n",
    "pd.DataFrame.from_dict(snpmaps,orient='index',columns=['ID', 'ALT']).to_csv('snpslist.txt',header=None, index=False,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "5f27a790-338d-4ba2-97b6-159a7fdfb5ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "uid = d.participant_id.unique()\n",
    "pd.DataFrame({'#FID':uid, 'IID':uid}).to_csv('idlist.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "65c8a9e3-4cfe-4b85-8bc0-2a56170b06c6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EXEC_TIME in sec: 153.507 : extract snps of interest from the PLINK2 AMP v2.5 files \n",
      "\n"
     ]
    }
   ],
   "source": [
    "plink_file_path = '/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/pfile/amp_v2.5_allchr'\n",
    "\n",
    "t = submitTerminal(plink2_cmd(f'{plink_file_path}', 'snps', f'--keep idlist.txt --extract snpslist.txt --export A --export-allele snpslist.txt'), message='extract snps of interest from the PLINK2 AMP v2.5 files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "a1d9a749-3270-4dcb-9b1a-a650f35a35ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# seems that all 5 variants successfully retrieved (sometimes, rsID are missing and can be missed by rsID based extraction)\n",
    "t = pd.read_csv('snps.raw', sep='\\t')\n",
    "t = t.rename(columns=\n",
    "    {'IID':'participant_id',\n",
    "     'rs76763715_C':'GBA_N370N', \n",
    "     'rs2230288_T':'GBA_E326K',\n",
    "     'rs104893877_T':'SNCA_A53T',\n",
    "     'rs34637584_A':'LRRK2_G2019S'}\n",
    ")\n",
    "t = t.drop(columns=['FID', 'PAT', 'MAT', 'SEX','PHENOTYPE'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "889454a9-711d-42b9-9c2d-85cde055a988",
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine with clinical data\n",
    "t2 = pd.merge(d, t, on='participant_id', how='left')\n",
    "t3 = pd.read_csv('/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/euro_king_pca_v2.5_July2021/genetic_ancestry_all_pca.csv')\n",
    "t3 = t3.rename(columns={'IID':'participant_id'}).drop(columns=['#FID'])\n",
    "df = pd.merge(t2, t3, on='participant_id', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4cb494c1-85e8-4788-a048-54e27d0c944c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5923, 38)"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "3ddaadb6-416d-4478-b39b-5615858e421f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATASET</th>\n",
       "      <th>LC</th>\n",
       "      <th>PD</th>\n",
       "      <th>PP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>visit_month</th>\n",
       "      <th>visit_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th>M0</th>\n",
       "      <td>533</td>\n",
       "      <td>1349</td>\n",
       "      <td>1827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <th>M12</th>\n",
       "      <td>80</td>\n",
       "      <td>556</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <th>M24</th>\n",
       "      <td>96</td>\n",
       "      <td>489</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <th>M36</th>\n",
       "      <td>95</td>\n",
       "      <td>391</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <th>M48</th>\n",
       "      <td>90</td>\n",
       "      <td>121</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "DATASET                  LC    PD    PP\n",
       "visit_month visit_name                 \n",
       "0           M0          533  1349  1827\n",
       "12          M12          80   556     5\n",
       "24          M24          96   489   129\n",
       "36          M36          95   391   118\n",
       "48          M48          90   121    44"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.pivot_table(index=['visit_month','visit_name'], columns='DATASET', values='participant_id', aggfunc='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "9f8f8d5c-9d86-4f79-8abd-07f68a77e310",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df.DATASET!='PP', ['participant_id', 'visit_name', 'visit_month', 'upsit_total_score',\n",
    "    'age_at_baseline', 'sex', 'study_arm', 'prodromal_category',\n",
    "       'diagnosis_at_baseline', 'diagnosis_latest',\n",
    "       'case_control_other_at_baseline',\n",
    "       'age_at_diagnosis', 'mds_updrs_part_i_summary_score',\n",
    "       'mds_updrs_part_ii_summary_score', 'mds_updrs_part_iii_summary_score',\n",
    "       'upd23b_clinical_state_on_medication',\n",
    "       'mds_updrs_part_iv_summary_score', 'updrs1_ment_behav_mood_score',\n",
    "       'updrs2_adl_score', 'updrs3_motor_examination_score',\n",
    "       'updrs4_therapy_complications_score', 'DATASET', 'GBA_N370N',\n",
    "       'GBA_E326K', 'SNCA_A53T', 'LRRK2_G2019S',\n",
    "       'InfPop','PC1', 'PC2', 'PC3',\n",
    "       'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10']].to_csv('UPSIT_genetics_LC_PD.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "be33a0c5-b028-4205-a519-93f70dc5b651",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['participant_id', 'visit_name', 'visit_month', 'upsit_total_score',\n",
       "       'age_at_baseline', 'sex', 'study_arm', 'prodromal_category',\n",
       "       'diagnosis_at_baseline', 'diagnosis_latest',\n",
       "       'case_control_other_at_baseline', 'case_control_other_latest',\n",
       "       'age_at_diagnosis', 'mds_updrs_part_i_summary_score',\n",
       "       'mds_updrs_part_ii_summary_score', 'mds_updrs_part_iii_summary_score',\n",
       "       'upd23b_clinical_state_on_medication',\n",
       "       'mds_updrs_part_iv_summary_score', 'updrs1_ment_behav_mood_score',\n",
       "       'updrs2_adl_score', 'updrs3_motor_examination_score',\n",
       "       'updrs4_therapy_complications_score', 'DATASET', 'GBA_N370N',\n",
       "       'GBA_E326K', 'SNCA_A53T', 'LRRK2_G2019S', 'InfPop', 'PC1', 'PC2', 'PC3',\n",
       "       'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "12964e22-0d65-4a27-8d04-12582a56663a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>visit_name</th>\n",
       "      <th>M0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DATASET</th>\n",
       "      <th>study_arm</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">PD</th>\n",
       "      <th>Disease Control</th>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Healthy Control</th>\n",
       "      <td>480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PD</th>\n",
       "      <td>809</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "visit_name                M0\n",
       "DATASET study_arm           \n",
       "PD      Disease Control   60\n",
       "        Healthy Control  480\n",
       "        PD               809"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.pivot_table(index=['DATASET', 'study_arm'], columns='visit_name', values='participant_id', aggfunc='count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9464c21a-5d30-4e91-9457-0198b8ffc403",
   "metadata": {},
   "source": [
    "# Get the genetic PCs for all AMP-PD samples + GUID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0198de50-6fa5-4fc4-a6bd-597a91751d0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demographics, enrollment category case-control status\n",
    "d2 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/Demographics.csv').drop_duplicates()\n",
    "d3 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/Enrollment.csv').drop_duplicates()\n",
    "d4 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/amp_pd_case_control.csv').drop_duplicates()\n",
    "t2 = pd.merge(d2[['participant_id', 'GUID',  'age_at_baseline', 'sex']], d3[['participant_id', 'study_arm', 'prodromal_category']], on='participant_id', how='left')\n",
    "t3 = pd.merge(t2, d4, on='participant_id',  how='left').drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "463c308e-502b-4546-8ee8-bd5e79ed2394",
   "metadata": {},
   "outputs": [],
   "source": [
    "# age at diagnosis (at visit_month==0)\n",
    "d5 = pd.read_csv('/data/CARD/PD/AMP-PD/Clinical/2021_v2_5release/clinical/PD_Medical_History.csv').drop_duplicates()\n",
    "d5 = d5[(pd.notna(d5.age_at_diagnosis))&(d5.visit_month==0)].copy()\n",
    "t4 = pd.merge(t3, d5[['participant_id', 'age_at_diagnosis']], on='participant_id',  how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "51b7c2b6-17b5-4ced-a507-8162a1763fff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# genetic PCs\n",
    "t3 = pd.read_csv('/data/CARD/PD/AMP-PD/Plink/2021_v2_5release/euro_king_pca_v2.5_July2021/genetic_ancestry_all_pca.csv')\n",
    "t3 = t3.rename(columns={'IID':'participant_id'}).drop(columns=['#FID'])\n",
    "df = pd.merge(t4, t3, on='participant_id', how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1e3a4c2c-c945-41c9-80f5-76f120801950",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add dataset\n",
    "df['DATASET'] = df.participant_id.str.slice(0,2) # LD, PP, PD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a36ebe9e-1978-44fa-86c2-ba6f772cb5ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"demographicsPCs.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "8bd343ed-6290-4102-b5f0-0a1216ee00ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['participant_id', 'GUID', 'age_at_baseline', 'sex', 'study_arm',\n",
       "       'prodromal_category', 'diagnosis_at_baseline', 'diagnosis_latest',\n",
       "       'case_control_other_at_baseline', 'case_control_other_latest',\n",
       "       'age_at_diagnosis', 'InfPop', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6',\n",
       "       'PC7', 'PC8', 'PC9', 'PC10', 'DATASET'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef9b0ffe-b4cf-44d9-aad5-88f4cbb3fe34",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.8",
   "language": "python",
   "name": "py3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
